freeipa/src/main/java/com/sequenceiq/freeipa/controller/FreeIpaV1Controller.java

```diff
@@ -342,4 +348,14 @@ public GenerateImageCatalogResponse generateImageCatalog(@ResourceCrn @NotEmpty
         String accountId = crnService.getCurrentAccountId();
         return imageCatalogGeneratorService.generate(environmentCrn, accountId);
     }
+
+    @Override
+    @InternalOnly
+    public OperationStatus upgradeCcmInternal(
+            @ValidCrn(resource = CrnResourceDescriptor.ENVIRONMENT) @ResourceCrn @TenantAwareParam @NotEmpty String environmentCrn,
+            @ValidCrn(resource = CrnResourceDescriptor.USER) @InitiatorUserCrn @NotEmpty String initiatorUserCrn) {
```

Comment:
you do not need both `@TenantAwaraParam` and `@InitiatorUserCrn` annotation, one of them is enough for an internal endpoint. If you want to use environmentCrn for internal call to pick up account ID, please remove initiatorUserCrn parameter, if you want initiatorUserCrn to pick up account ID and use as an user CRN, please remove `@TenantAwareParam` annotation.

Reply:
Oh, got it. They are expected to share the same account ID, and the `initiatorUserCrn` is also desirable, so I will then remove the `@TenantAwareParam` annotation (but still keep the `environmentCrn` argument).

Reply:
you have not removed id (yet)

Reply:
Done.